From 7473a83b705b26b3bc1760d6fb533409e99bf1c5 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sun, 29 Jul 2018 20:19:19 +0200
Subject: [PATCH 10/13] LineageParts: Disable dark mode on light design

* If light design is chosen it doesn't make sense to choose the preferred
  dark mode
* Disable the preference in that case

Change-Id: I0ef64afe91abd52572906c3eb280fec2a4592cad
---
 .../lineageparts/style/StylePreferences.java       | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/src/org/lineageos/lineageparts/style/StylePreferences.java b/src/org/lineageos/lineageparts/style/StylePreferences.java
index 2907fae..7df3fbe 100644
--- a/src/org/lineageos/lineageparts/style/StylePreferences.java
+++ b/src/org/lineageos/lineageparts/style/StylePreferences.java
@@ -48,7 +48,6 @@ import org.lineageos.lineageparts.style.util.UIUtils;
 import java.util.Arrays;
 import java.util.List;
 
-import lineageos.preference.LineageSystemSettingListPreference;
 import lineageos.providers.LineageSettings;
 import lineageos.style.StyleInterface;
 import lineageos.style.Suggestion;
@@ -61,6 +60,7 @@ public class StylePreferences extends SettingsPreferenceFragment {
 
     private Preference mStylePref;
     private Preference mAccentPref;
+    private Preference mDarkPref;
 
     private List<Accent> mAccents;
 
@@ -88,9 +88,9 @@ public class StylePreferences extends SettingsPreferenceFragment {
         mAccentPref.setOnPreferenceClickListener(this::onAccentClick);
         setupAccentPref();
 
-        LineageSystemSettingListPreference darkPref = (LineageSystemSettingListPreference)
-                findPreference("berry_dark_overlay");
-        darkPref.setOnPreferenceChangeListener(this::onDarkChange);
+        mDarkPref = findPreference("berry_dark_overlay");
+        mDarkPref.setOnPreferenceChangeListener(this::onDarkChange);
+        setDarkStyleEnabled(mInterface.getGlobalStyle());
 
         Preference automagic = findPreference("style_automagic");
         automagic.setOnPreferenceClickListener(p -> onAutomagicClick());
@@ -276,10 +276,16 @@ public class StylePreferences extends SettingsPreferenceFragment {
         // selection dialog to be dismissed gracefully
         new Handler().postDelayed(() -> mInterface.setGlobalStyle(value, mPackageName), 500);
 
+        setDarkStyleEnabled(value);
         setStyleIcon(value);
+
         return true;
     }
 
+    private void setDarkStyleEnabled(int value) {
+        mDarkPref.setEnabled(value != StyleInterface.STYLE_GLOBAL_LIGHT);
+    }
+
     private void setStyleIcon(int value) {
         int icon;
         switch (value) {
-- 
2.17.1

